build.sbt

```diff
@@ -2297,8 +2372,11 @@ lazy val `engine-runner` = project
     assembly / test := {},
     assembly / assemblyOutputPath := file("runner.jar"),
     assembly / assemblyExcludedJars :=
-      JPMSUtils.filterTruffleAndGraalArtifacts(
-        (Compile / fullClasspath).value
+      JPMSUtils.filterArtifacts(
+        (Compile / fullClasspath).value,
+        "graalvm",
+        "truffle",
+        "helidon"
```

Comment:
This ensures that there are no helidon classes in `runner.jar`. This means that all the helidon classes should be present in helidon modules inside the `component` directory. That is, all the helidon classes should be present on the module boot layer. Thus, modify also the `CLASS_DELEGATION_PATTERNS` in https://github.com/enso-org/enso/blob/develop/engine/runtime/src/main/java/org/enso/ClassLoaderConstants.java#L15 to contain helidon classes as well. If you add the `org.helidon` prefix there, our boot class loader ensures that all the classes with `org.helidon` prefix will be loaded from the module boot layer (i.e. from JARs in `component` directory)

Reply:
By the way, I believe that if you add `org.helidon` to the `CLASS_DELEGATION_PATTERNS` (https://github.com/enso-org/enso/blob/develop/engine/runtime/src/main/java/org/enso/ClassLoaderConstants.java#L15), it won't be necessary to remove helidon classes from `runner.jar` anymore.

Reply:
I have just [noticed](https://github.com/enso-org/enso/pull/9862/files#r1598079038) that the change is already there. Sorry for the fuzz.